GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitCommandsTests.cs

```diff
@@ -45,6 +46,48 @@ public void StatusTest()
             this.ValidateGitCommand("status");
         }
 
+        [TestCase]
+        public void StatusWithGitTraceTest()
+        {
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+
+            string tracePath = Path.Combine(this.Enlistment.EnlistmentRoot, "trace-log.txt");
+            environmentVariables["GIT_TRACE"] = tracePath;
+            this.ValidateGitCommand(environmentVariables, "status");
+
+            this.FileSystem.FileExists(tracePath).ShouldBeTrue();
+
+            string lineToFind = "trace: built-in: git status";
+            int count = 0;
+
+            foreach (string line in this.FileSystem.ReadAllText(tracePath).Split('\n'))
+            {
+                if (line.IndexOf(lineToFind) >= 0)
+                {
+                    count++;
+                }
+            }
+
+            // The GIT_TRACE output is appended to the file, so we should have
+            // one copy for each run: normal and virtualized.
+            count.ShouldEqual(2);
+        }
+
+        [TestCase]
+        public void StatusWithGitTraceInvalidCharactersTest()

```

Comment:
Do we already have a test where `status` is run with a valid value for "GIT_TRACE" (that is not a file)? If not, can\should we add one?

Reply:
Hm. Perhaps I could do it with `GIT_TRACE=2` to pipe to stderr, since if we test with `GIT_TRACE=1` the output will be different.